File: Java/src/_0000_classicalProblems/mylinkedlist/MyLinkedList.java
package _0000_classicalProblems.mylinkedlist;

/**
 * Date: 2020/9/4 18:08
 * Content:
 */
public class MyLinkedList {
    private ListNode head;  // 链表头

    public MyLinkedList() {
        head = null;
    }

    public MyLinkedList(int[] nums) {
        this();
        head = createLinkedList(nums);
    }

    // 创建链表
    public ListNode createLinkedList(int[] nums) {
        if (0 == nums.length)
            return null;

        ListNode head = new ListNode(nums[0]);

        ListNode curNode = head;
        for (int i = 1; i < nums.length; i++) {
            curNode.next = new ListNode(nums[i]);
            curNode = curNode.next;
        }

        return head;
    }

    // 打印链表
    public void printLinkedList(ListNode head) {
        ListNode curNode = head;
        while (null != curNode) {
            System.out.print(curNode.val);
            System.out.print("->");
            curNode = curNode.next;
        }
        System.out.println("NULL");
    }
}

File: Python/_0024_Swap_Nodes_in_Pairs/MyLinkedList.py
# coding=utf-8


# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class MyLinkedList:
    # head = MyLinkedList().createLinkedList([1, 2, 3])
    def __int__(self):
        pass

    def createLinkedList(self, nums):
        if len(nums) == 0:
            return None

        head = ListNode(nums[0])
        cur = head
        for i in range(1, len(nums)):
            cur.next = ListNode(nums[i])
            cur = cur.next

        return head

    def printLinkedList(self, head):
        cur = head
        while cur:
            print(cur.val, "->", sep="", end="")  # sep取消空格，end取消回车
            cur = cur.next
        print("NULL")
